producer/writer/shard_writer.go

```diff
@@ -249,6 +260,17 @@ func (w *replicatedShardWriter) Close() {
 	}
 }
 
+func (w *replicatedShardWriter) QueueSize() int {
```

Comment:
Is the total queue size sufficient to allow effective diagnostics?

Reply:
yeah